tools/autograd/derivatives.yaml

```diff
@@ -1678,7 +1678,12 @@
   result: auto_linear
 
 - name: lift(Tensor self) -> Tensor
-  self: not_implemented("lift")
+  self: grad
+  result: auto_linear
+
+- name: lift_fresh(Tensor(a) self) -> Tensor(a)
```

Comment:
`lift_fresh` only intended to be called when self is a Tensor materialized out of nowhere right? Wouldn't it always have requires_grad=False then?

Reply:
yes... but when I ran the xla tests they errored because of a missing `lift_fresh` derivative, I guess I will need to investigate more

Reply:
It's a matter of semantics. When a mode is applied, alias is expected to get tensor inputs that already have had all of the mode wrapping applied. `lift_fresh` is special: it is called specifically when (1) we know no mode wrapping could have been applied on the input (because we disabled all modes when allocating it) and (2) we know that the input tensor was freshly allocated (so it is sound to return an alias and mutate it.)